OneCommunityGlobal/HighestGoodNetworkApp
# XiaoW_new timer refactor

# Description
This is the refactored new timer frontend, and the related components were based on PR [#1021](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1021#issue-1811282749). 
The new timer is based on Websocket connection communication and it is a count down timer instead of stop watch timer like the old timer.

## Related PRS (if any):
This frontend PR is related to the [#575](https://github.com/OneCommunityGlobal/HGNRest/pull/575#issue-1948845268) backend PR.
To test this frontend PR you need to checkout the  [#575](https://github.com/OneCommunityGlobal/HGNRest/pull/575#issue-1948845268) backend PR.

## How to test:
1. check into current branch
2. check into backend branch
3. do `npm install` and `npm run start:local` to run this PR locally
4. log as any role
5. go to dashboard
6. verify function the new timer is working as intended (set/add/remove goal, start, pause, stop and log time...), try to break it.

## Note:
1. The time range it can be set is between 15mins and 10 hours. 
2. Refresh will cause the Websocket connection to close, which would lead to a forced pause action
3. Be aware of browser memory saver (may kick in after 2 hours of inactive for chrome), the timer is designed to pause and keep the elapsed time when Websocket connection is lost.

## Updates:
1. After discussing with Jae, the time range now is 1 min to 5 hours;
2. the alignment issue is fixed.
3. the gong audio now stops after TimeIsOverModal is closed
4. Part of the UI changes.
